python/tk_vred/menu_generation.py

```diff
@@ -39,9 +39,8 @@ def favourites(self):
         The menu favourites, in sorted order by name.
         """
 
-        if self._menu_favourites is None:
-            self._menu_favourites = self._engine.get_setting("menu_favourites", [])
-            self._menu_favourites.sort(key=lambda f: f["name"])
+        self._menu_favourites = self._engine.get_setting("menu_favourites", [])
+        self._menu_favourites.sort(key=lambda f: f["name"])
```

Comment:
This is safe to rebuild each time, but I think this issue stems higher up from not clearing the menu favourites when we clear the menu when rebuilding the menu (e.g. on a context change).

We should instead just clear the menu favourites in `clean_menu` method
```
self._menu_favourites = None
```
and this should resolve the issue.

Reply:
Hey @staceyoue , I tried that and it does not seem to work.

Reply:
Ok, I think we should still set the menu favourites to None in the clean method regardless.

If this does not work, then I assume we are calling the `create_menu` method with `clean_menu=False`, which does not reset the menu favourites.

In this case, if we are rebuilding the menu favourites each time, we should change the `menu_favourites` property to a method:
```
def get_menu_favourites(self):
    // fetch the menu favourites from the settings
```

Reply:
@staceyoue my environment had an issue so my test above was invalid. :(